File: java-web-spring-boot-3/src/main/java/com/vndevteam/javawebspringboot3/infrastructure/exception/error/ErrorResponseDto.java
package com.vndevteam.javawebspringboot3.infrastructure.exception.error;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.ZonedDateTime;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ErrorResponseDto {
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
    private ZonedDateTime timestamp;
    private String code;
    private String message;
    private List<ErrorDetail> errors;
    private String trace;
}

File: kotlin-web-spring-boot-3/src/main/kotlin/com/vndevteam/kotlinwebspringboot3/infrastructure/exception/error/ErrorResponseDto.kt
package com.vndevteam.kotlinwebspringboot3.infrastructure.exception.error

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.ZonedDateTime

class ErrorResponseDto(
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX") var timestamp: ZonedDateTime?,
    var code: String?,
    var message: String?,
    var errors: List<ErrorDetail>?,
    var trace: String?
)
